tests/5.1/masked/test_masked.F90

```diff
@@ -0,0 +1,54 @@
+!/===--- test_masked.F90 -------------------------------------===//
+!
+! OpenMP API Version 5.0 Nov 2018
+!
+! This test utilizes the masked construct to ensure that a section of code
+! only runs on a certain thread. When the filter clause is not present, the
+! primary thread runs the structured block.
+!
+!/===----------------------------------------------------------------------===//
+
+#include "ompvv.F90"
+
+#define N 1024
+
+PROGRAM test_masked
+  USE iso_fortran_env
+  USE ompvv_lib
+  USE omp_lib
+  implicit none
+
+  OMPVV_TEST_OFFLOADING
+
+  OMPVV_TEST_VERBOSE(masked() .ne. 0)
+
+  OMPVV_REPORT_AND_RETURN()
+
+CONTAINS
+  INTEGER FUNCTION masked()
+    INTEGER:: errors = 0
+    INTEGER:: total = 10
+    INTEGER:: ct = 0
+    INTEGER, DIMENSION(N):: x
+    INTEGER:: i
+
+    OMPVV_INFOMSG("test_masked")
+
+    DO i = 0, N
+       x(i) = 0
+    END DO
+    threads = omp_get_num_threads()
+
+    !$omp parallel num_threads(threads)
+    DO WHILE (total>0)
+        !$omp masked
+            OMPVV_TEST_AND_SET_VERBOSE(errors, omp_get_thread_num() .ne. 0) ! primary thread
+            ct = ct + 1
+            total = total - 1
+        !$omp end masked
+    END DO
```

Comment:
With `parallel` either a `BLOCK ... END BLOCK` or an end-directive is required, e.g.:
```suggestion
    END DO
    !$omp end parallel
```